content/how-to/bpmn/tune-performance.md

```diff
@@ -53,3 +51,20 @@ A strategy to debug and minimize necessary computation is to break transformatio
 If you use Visual Studio Code, consider the [`jsonata-language`](https://marketplace.visualstudio.com/items?itemName=bigbug.vscode-language-jsonata) extension.
 Similar to a Jupyter notebook, the extension provides an interactive environment to write JSONata expressions and to pass the outputs from one expression into the input of another.
 Besides its benefit for monitoring performance, we have used it to incrementally build complex JSONata in a way that we can  document and share (in the style of literate programming).
+
+## Manage the process context size
+
+{{< notice "note" >}}
+The max size of the process variable context comes from the default max payload size of NATS Jetstreams.
+To increase this size, change your NATS configuration.
+{{< /notice >}}
+
+By default, the size of the {{< abbr "process variable context" >}}) is 1MB.
+If the sum size of all variables exceeds this limit, the BPMN process exits.
+
+Be mindful of the overall size of your variables, especially when outputting to new variables.
+For example, imagine an initial JSON payload, `data`, that is 600MB.
+If a JSONata task slightly modifies and outputs it to a new variable, `data2`, the process variable context will exceed 1000MB and the BPMN process will exit.
+
+To work around this constraint, you can save memory by mutating variables.
+That is, instead of outputting a new variable, you can output the transformed payload to the original variable name.
```

Comment:
You can also use the `responseTransformExpression` to minimize the data returned by large requests in GraphQL or REST nodes. Only the output of the transform will be stored, not the original response.

Reply:
![image](https://github.com/libremfg/libremfg.github.io/assets/47385188/6f7ee2d8-a34b-4509-b603-6fa24d7343e5)

This right? But, just to be sure―any variables that came before will still occupy memory in the context space. It's a good tip though to keep the blast radius down.

Reply:
The way it works is that the original response never goes on the variable context if there is a response transform expression, only the output of that expression.

`var` above refers to where the relevant **one** of them will go on completion of the job.